### PR TITLE
feat: #60 Adding support for ring deployments

### PR DESCRIPTION
**Pull request by juanultimate**
It adds support for ring deployments. 
Using the Azure App configuration service, the time-tracker-ui is able to enable/disable features to certain users or group of users (  aka role).

### CODE REVIEW

File: src/app/modules/login/services/azure.ad.b2c.service.spec.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+import { UserAgentApplication } from 'msal';
+
 import { AzureAdB2CService } from './azure.ad.b2c.service';
-import { UserAgentApplication, Account } from 'msal';
 
 describe('AzureAdB2CService', () => {
   let service: AzureAdB2CService;
 
-  let account: Account;
+  let account;
```
Comment (enriquezrene) [inline src/app/modules/login/services/azure.ad.b2c.service.spec.ts:9]: do we have a strong reason to remove the data type from here?

File: src/app/modules/login/services/azure.ad.b2c.service.ts [hunk 2]
```diff
@@ -55,4 +56,13 @@ export class AzureAdB2CService {
   getBearerToken(): string {
     return sessionStorage.getItem('msal.idtoken');
   }
+
+  getUserEmail(): string {
+    console.log(this.msal.getAccount());
```
Comment (enriquezrene) [inline src/app/modules/login/services/azure.ad.b2c.service.ts:61]: I think you forgot to remove this, did not you?

File: src/app/modules/reports/pages/reports.component.ts [hunk 3]
```diff
@@ -1,15 +1,9 @@
...
   styleUrls: ['./reports.component.scss']
 })
-export class ReportsComponent implements OnInit {
-
-  constructor() { }
-
-  ngOnInit(): void {
-  }
-
+export class ReportsComponent {
```
Comment (enriquezrene) [inline src/app/modules/reports/pages/reports.component.ts:8]: Thanks for this

File: src/app/modules/shared/feature-toggles/feature-toggle-manager.service.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  constructor(private featureToggleProvider: FeatureToggleProvider) { }
+
+  public isToggleEnabled(toggleName: string, toggleLabel?: string): Observable<boolean> {
+    return this.featureToggleProvider.getFeatureToggle(toggleName, toggleLabel).pipe(
+      map(featureToggle => featureToggle.enabled)
+    );
+  }
+
+  public isToggleEnabledInContext(toggleName: string, toggleLabel?: string): Observable<boolean> {
+    return this.featureToggleProvider.getFeatureToggle(toggleName, toggleLabel).pipe(
```
Comment (enriquezrene) [inline src/app/modules/shared/feature-toggles/feature-toggle-manager.service.ts:21]: In what context? Could you please give me a brief explanation about this?

File: src/app/modules/shared/feature-toggles/filters/feature-filter-types.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export enum FeatureFilterTypes {
+  TARGETING = 'Microsoft.Targeting'
```
Comment (enriquezrene) [inline src/app/modules/shared/feature-toggles/filters/feature-filter-types.ts:2]: What other targeting values we might have?

File: src/app/modules/shared/feature-toggles/filters/targeting/targeting-feature-filter.model.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { FeatureFilterModel } from '../feature-filter.model';
+import { TargetingFilterParameters } from './targeting-feature-filter-parameters';
+import { TargetingFilterAppContext } from './targeting-filter-app-context';
+
+export class TargetingFeatureFilterModel implements FeatureFilterModel {
+
+  name = 'Microsoft.Targeting';
```
Comment (enriquezrene) [inline src/app/modules/shared/feature-toggles/filters/targeting/targeting-feature-filter.model.ts:7]: Did not you define a enum for this value? `FeatureFilterTypes`

File: src/app/modules/login/services/azure.ad.b2c.service.spec.ts (hunk 1 above)
Comment (juanultimate) [reply src/app/modules/login/services/azure.ad.b2c.service.spec.ts:9]: I do not. Let me bring it back.

File: src/app/modules/login/services/azure.ad.b2c.service.ts (hunk 2 above)
Comment (juanultimate) [reply src/app/modules/login/services/azure.ad.b2c.service.ts:61]: Yep. Good catch!!

File: src/app/modules/shared/feature-toggles/filters/targeting/targeting-feature-filter.model.ts (hunk 6 above)
Comment (juanultimate) [reply src/app/modules/shared/feature-toggles/filters/targeting/targeting-feature-filter.model.ts:7]: My bad. Let me update it

File: src/app/modules/shared/feature-toggles/filters/feature-filter-types.ts (hunk 5 above)
Comment (juanultimate) [reply src/app/modules/shared/feature-toggles/filters/feature-filter-types.ts:2]: I would say TimeWindow, UserPercentage, LocationBased, TenantBased or any other custom filter we might have in mind.

**Review by enriquezrene — APPROVED** [review]

**Pull request merged**